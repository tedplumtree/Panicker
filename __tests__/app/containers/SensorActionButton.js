import configureStore from 'redux-mock-store';
import {
  mapStateToProps,
  mapDispatchToProps,
} from '../../../app/containers/SensorActionButton';

const mockStore = configureStore([]);

describe('mapStateToProps', () => {
  const dummyOwnProps = {
    activateIcon: 'dummy-activate-icon',
    deactivateIcon: 'dummy-deactivate-icon',
  };

  describe('when sensor is active', () => {
    const fakeStore = mockStore({ sensor: { active: true } });

    it('passes the correct props', () => {
      const props = mapStateToProps(fakeStore.getState(), dummyOwnProps);

      expect(props).toMatchSnapshot();
    });
  });

  describe('when sensor is inactive', () => {
    const fakeStore = mockStore({ sensor: { active: false } });

    it('passes the correct props', () => {
      const props = mapStateToProps(fakeStore.getState(), dummyOwnProps);

      expect(props).toMatchSnapshot();
    });
  });
});

describe('mapDispatchToProps', () => {
  it('passes correct props to connected component', () => {
    const props = mapDispatchToProps(mockStore().dispatch);

    expect(props).toMatchSnapshot();
  });

  it('passes prop onPress that dispatches correct action', () => {
    const fakeStore = mockStore();
    const props = mapDispatchToProps(fakeStore.dispatch);

    props.onPress();

    expect(fakeStore.getActions()).toMatchSnapshot();
  });
});
