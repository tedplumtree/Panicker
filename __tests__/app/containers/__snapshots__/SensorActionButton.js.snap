// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapDispatchToProps passes correct props to connected component 1`] = `
Object {
  "onPress": [Function],
}
`;

exports[`mapDispatchToProps passes prop onPress that dispatches correct action 1`] = `
Array [
  Object {
    "type": "TOGGLE_SENSOR",
  },
]
`;

exports[`mapStateToProps when sensor is active passes the correct props 1`] = `
Object {
  "icon": "dummy-deactivate-icon",
}
`;

exports[`mapStateToProps when sensor is inactive passes the correct props 1`] = `
Object {
  "icon": "dummy-activate-icon",
}
`;
